fundraiser/tests/harness.rs

```diff
@@ -0,0 +1,1350 @@
+mod utils;
+
+use fuels::{
+    signers::Signer,
+    tx::{AssetId, ContractId},
+};
+
+use utils::{
+    abi_calls::{
+        campaign, campaign_count, campaign_info, cancel_campaign, claim_pledges, create_campaign,
+        pledge, pledge_count, pledged, total_campaigns, unpledge,
+    },
+    test_helpers::{mint, setup},
+    Identity,
+};
+
+// TODO: Until the SDK supports block manipulation changing tests may break them because of the
+//       specifically selected block deadlines so your test might be correct but the deadline is
+//       messing up the test
+//
+//  - claim_pledges
+//      - revert_when_claiming_before_deadline (need SDK to manipulate block height)
+//  - pledges
+//      - revert_when_pledging_after_deadline (need SDK to manipulate block height)
+//
+//      When logging is deserialized in the SDK, check logs are correct
```

Comment:
@Voxelot Here's an example of the need for the block manipulation/configuration thing we talked about recently. More specifically: https://github.com/FuelLabs/sway-applications/pull/67/files#diff-d3c232d3492ac06233511f72153a04c8e079fe6c141469787897a2b712af319dR460. 

Tagging you so you have some more motivation/context for that feature; it'll require some major SDK + client coordination to get this out.

Reply:
Should we be using timestamps instead of block numbers? I know there were some spec discussions about this, but I think timestamps would be safer for dapps as well as better for testing imo

Reply:
Timestamps should be used exclusively for dealing with the passing of time. That being said, given that there's no real money involved yet, block number is a drop-in replacement that serves the same purpose.